system-contracts/test/MsgValueSimulator.spec.ts

```diff
@@ -0,0 +1,194 @@
+    const contractBalanceBefore = await l2EthToken.balanceOf(mockContract.address);
+
+    await expect(
+      extraAbiCaller.fallback({
+        data: encodeExtraAbiCallerCalldata(
+          messageValueSimulator.address,
+          BigNumber.from(0),
+          [value.toString(), mockContract.address, "0"],
+          "0x"
+        ),
+      })
+    )
+      .to.emit(mockContract, "Called")
+      .withArgs(value, "0x");
+
+    const balanceAfter = await l2EthToken.balanceOf(msgSender);
+    expect(balanceBefore.sub(value)).to.equal(balanceAfter);
+    const contractBalanceAfter = await l2EthToken.balanceOf(mockContract.address);
+    expect(contractBalanceBefore.add(value)).to.equal(contractBalanceAfter);
+  });
+
+  it("send 1 wei", async () => {
+    const value = 1;
+    const balanceBefore = await l2EthToken.balanceOf(msgSender);
+    const contractBalanceBefore = await l2EthToken.balanceOf(mockContract.address);
+
+    await expect(
+      extraAbiCaller.fallback({
+        data: encodeExtraAbiCallerCalldata(
+          messageValueSimulator.address,
+          BigNumber.from(0),
+          [value.toString(), mockContract.address, "0"],
+          "0x"
+        ),
+      })
+    )
+      .to.emit(mockContract, "Called")
+      .withArgs(value, "0x");
+
+    const balanceAfter = await l2EthToken.balanceOf(msgSender);
+    expect(balanceBefore.sub(value)).to.equal(balanceAfter);
+    const contractBalanceAfter = await l2EthToken.balanceOf(mockContract.address);
+    expect(contractBalanceBefore.add(value)).to.equal(contractBalanceAfter);
+  });
+
+  it("send 2^127 wei", async () => {
+    const value = BigNumber.from(2).pow(127);
+    const balanceBefore = await l2EthToken.balanceOf(msgSender);
+    const contractBalanceBefore = await l2EthToken.balanceOf(mockContract.address);
+
+    await expect(
+      extraAbiCaller.fallback({
+        data: encodeExtraAbiCallerCalldata(
+          messageValueSimulator.address,
+          BigNumber.from(0),
+          [value.toString(), mockContract.address, "0"],
+          "0x"
+        ),
+      })
+    )
+      .to.emit(mockContract, "Called")
+      .withArgs(value, "0x");
+
+    const balanceAfter = await l2EthToken.balanceOf(msgSender);
+    expect(balanceBefore.sub(value)).to.equal(balanceAfter);
+    const contractBalanceAfter = await l2EthToken.balanceOf(mockContract.address);
+    expect(contractBalanceBefore.add(value)).to.equal(contractBalanceAfter);
+  });
+
+  it("revert with reentry", async () => {
+    const balanceBefore = await l2EthToken.balanceOf(msgSender);
+    const contractBalanceBefore = await l2EthToken.balanceOf(mockContract.address);
+
+    await expect(
+      extraAbiCaller.fallback({
+        data: encodeExtraAbiCallerCalldata(
+          messageValueSimulator.address,
+          BigNumber.from(0),
+          ["0x1", messageValueSimulator.address, "0"],
+          "0x"
+        ),
+      })
+    ).to.be.reverted;
```

Comment:
Can you check the revert reason?

Reply:
When I keep getting `AssertionError: Expected transaction to be reverted with reason 'c', but it reverted with a custom error`. Do you know how could I assert this custom error?

When not catching I get just a `UNPREDICTABLE_GAS_LIMIT`.

Reply:
It was a bug in the extra abi caller contract, but now should be fixed in dev